netket/nn/activation.py

```diff
@@ -48,6 +48,21 @@
 from jax.numpy import cosh
 from jax.numpy import sinh
 
+from netket.jax import HashablePartial
+
+
+@HashablePartial
+def reim(f):
+    # Modifies a non-linearity to act seperately on the real and imaginary parts
+    def cf(x):
+        sqrt2 = jnp.sqrt(jnp.array(2, dtype=x.real.dtype))
+        if jnp.iscomplexobj(x):
+            return jax.lax.complex(f(sqrt2 * x.real), f(sqrt2 * x.imag)) / sqrt2
+        else:
+            return f(x)
+
+    return cf
```

Comment:
that's how to use it.

```suggestion
    def reim_activation(f, x):
        sqrt2 = jnp.sqrt(jnp.array(2, dtype=x.real.dtype))
        if jnp.iscomplexobj(x):
            return jax.lax.complex(f(sqrt2 * x.real), f(sqrt2 * x.imag)) / sqrt2
        else:
            return f(x)

    return HashablePartial(reim_activation, f)
```